src/Core/Services/BuildRequestStateMiddleware.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.DataApiBuilder.Core.Authorization;
+using HotChocolate.Execution;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using RequestDelegate = HotChocolate.Execution.RequestDelegate;
+
+/// <summary>
+/// This request middleware will build up our request state and will be invoked once per request.
+/// </summary>
+public sealed class BuildRequestStateMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public BuildRequestStateMiddleware(RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    /// <summary>
+    /// Middleware invocation method which attempts to replicate the
+    /// http context's "X-MS-API-ROLE" header/value to HotChocolate's request context.
+    /// </summary>
+    /// <param name="context">HotChocolate execution request context.</param>
+    public async ValueTask InvokeAsync(IRequestContext context)
+    {
+        if (context.ContextData.TryGetValue(nameof(HttpContext), out object? value) &&
+            value is HttpContext httpContext)
+        {
+            // Because Request.Headers is a NameValueCollection type, key not found will return null and not an exception.
+            StringValues clientRoleHeader = httpContext.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER];
```

Comment:
Yes, because `httpContext.Request.Headers` is a collection where key: string, value: StringValues:
[ICollection](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1)<[KeyValuePair](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.keyvaluepair-2)<[String](https://learn.microsoft.com/en-us/dotnet/api/system.string),[StringValues](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringvalues)>>

The contents of the client role header are validated in ClientRoleHeaderAuthorizationMiddleware which is executed prior to the HotChocolate middleware pipeline. 
An invalid StringValues object is described below (comments are already present in the class file `ClientRoleHeaderAuthorizationMiddleware.cs`) and when encountered results in an immediate 403 Forbidden response.

```csharp
// When count = 0, the clientRoleHeader is absent on requests.
// Consequentially, anonymous requests must specifically set
// the clientRoleHeader value to Anonymous.

// When count > 1, multiple header fields with the same field-name
// are present in a message, but are NOT supported, specifically for the client role header.
// Valid scenario per HTTP Spec: http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
// Discussion: https://stackoverflow.com/a/3097052/18174950
```

Reply:
My comment was if it should be a `nullable` type like **StringValues?** <- notice the ending `?`. I ask this because the comment says: 

> // Because Request.Headers is a NameValueCollection type, key not found will return null and not an exception.

Should the comment instead say, a key not found will return an empty NameValueCollection with a count of 0?

Reply:
Sorry for missing that. `StringValues` doesn't need to be nullable because a keynot found will result in returning `StringValues.Empty` , not `null`.
updated the comment:
>             // Because Request.Headers is a NameValueCollection type, key not found will return StringValues.Empty and not an exception.